peekingduck/pipeline/nodes/model/yolov6_fam/yolov6_files/detector.py

```diff
@@ -0,0 +1,376 @@
+# Copyright 2022 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Detector module to predict object bbox from an image using YOLOv6.
+
+Changes:
+1. Removed redundant stuff such as tensorrt checks for now;
+2. Moved NUM_CHANNELS to config.
+3. Added model_switch from YOLOv6 to switch layers to deploy mode.
+"""
+
+from __future__ import annotations
+
+import logging
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import numpy as np
+import torch
+
+
+from peekingduck.pipeline.nodes.model.yolov6_fam.yolov6_files.models.yolo import YOLOv6
+from peekingduck.pipeline.nodes.model.yolov6_fam.yolov6_files.utils.torch_utils import (
+    fuse_model,
+)
+from peekingduck.pipeline.utils.bbox.transforms import xyxy2xyxyn
+
+
+from peekingduck.pipeline.nodes.model.yolov6_fam.yolov6_files.layers.common import (
+    RepVGGBlock,
+)
+from peekingduck.pipeline.nodes.model.yolov6_fam.yolov6_files.data.data_augment import (
+    letterbox,
+)
+from peekingduck.pipeline.nodes.model.yolov6_fam.yolov6_files.utils.nms import (
+    non_max_suppression,
+)
+
+from peekingduck.pipeline.nodes.model.yolov6_fam.yolov6_files.configs.config import (
+    ModelParams,
+)
+
+
+class Detector:  # pylint: disable=too-many-instance-attributes
+    """Object detection class using YOLOv6 to predict object bboxes.
+
+    Attributes:
+        logger (logging.Logger): Events logger.
+        config (Dict[str, Any]): YOLOv6 node configuration.
+        model_dir (pathlib.Path): Path to directory of model weights files.
+        device (torch.device): Represents the device on which the torch.Tensor
+            will be allocated.
+        half (bool): Flag to determine if half-precision should be used.
+        yolov6 (YOLOv6): The YOLOv6 model for performing inference.
+    """
+
+    # this is congruence to yolov6's config model params
+    model_params: type[ModelParams]
+    stride: int
```

Comment:
These two are not inside the `__init__` arguments, I thought it might be good to denote the type for these two variables as well. I could also define them as such:

```
self.stride: int = int(self.yolov6.stride.max())
```

instead of the more verbose way.

Reply:
I think so far we have been doing it in the less verbose way as you have mentioned in the comment [ref](https://github.com/aisingapore/PeekingDuck/blob/main/peekingduck/pipeline/nodes/input/visual.py#L158-L166)

Reply:
Noted, and changed.